File: ios/ReactNativeCurrencyLocalizationModule.swift
import ExpoModulesCore

public class ReactNativeCurrencyLocalizationModule: Module {
  public func definition() -> ModuleDefinition {
    Name("ReactNativeCurrencyLocalization")

    Function("getLocalizedCurrencyName") { (locale: String, currencyCode: String) -> [String : String] in
      return [
        "name": Locale(identifier: locale).localizedString(forCurrencyCode: currencyCode) ?? currencyCode,
        "symbol": NSLocale(localeIdentifier: currencyCode).displayName(forKey: .currencySymbol, value: currencyCode) ?? currencyCode
      ]
    }
  }
}

File: android/src/main/java/expo/community/modules/currencylocalization/ReactNativeCurrencyLocalizationModule.kt
package expo.community.modules.currencylocalization

import java.util.*
import expo.modules.kotlin.modules.Module
import expo.modules.kotlin.modules.ModuleDefinition

class ReactNativeCurrencyLocalizationModule : Module() {
  override fun definition() = ModuleDefinition {
    Name("ReactNativeCurrencyLocalization")

    Function("getLocalizedCurrency") { locale: String, currencyCode: String ->
      val localeObject = if (locale.contains("-")) {
        Locale(locale.split('-')[0], locale.split('-')[1])
      } else {
        Locale(locale)
      }
      val currency: Currency = Currency.getInstance(currencyCode)
      return@Function mapOf(
        "name" to currency.getDisplayName(localeObject),
        "symbol" to currency.getSymbol(localeObject)
      )
    }
  }
}